### PR TITLE
core/tracker: detect inconsistent partial signed data

### PR DESCRIPTION
**Pull request by corverroos**
Track inconsistent partial signature data in tracker.

- Distinguish between insufficient and inconsistent partial signatures when duty failed in parsigdb.
- Log and instrument when multiple partial signed datas detected.

category: feature
ticket: #1342

### CODE REVIEW

File: core/tracker/tracker.go [hunk 1]
```diff
@@ -559,105 +636,131 @@ func (t *Tracker) ConsensusEvent(ctx context.Context, duty core.Duty, data core.
...
+	} else {
+		log.Warn(ctx, "Inconsistent partial signed data", nil,
+			z.Any("data", parsigMsgs))
+	}
+}
+
+// expectInconsistentParSigs returns true if the duty type is expected to sometimes
+// produce inconsistent partial signed data.
+func expectInconsistentParSigs(duty core.DutyType) bool {
+	return duty == core.DutySyncMessage || duty == core.DutySyncContribution
```
Comment (xenowits) [inline core/tracker/tracker.go:765]: shouldn't this be
```suggestion
	return duty == core.DutySyncMessage || duty == core.DutyPrepareSyncContribution
```
??

File: core/tracker/tracker.go [hunk 2]
```diff
@@ -287,15 +327,49 @@ func analyseDutyFailed(duty core.Duty, allEvents map[core.Duty][]event) (bool, c
 	return true, comp, msg
 }
 
+// analyseParSigs returns a mapping partial signed data messages by peers (share index).
```
Comment (xenowits) [inline core/tracker/tracker.go:330]:
```suggestion
// analyseParSigs returns a mapping of partial signed data messages by peers (share index).
```

File: core/tracker/tracker.go [hunk 3]
```diff
@@ -287,15 +327,49 @@ func analyseDutyFailed(duty core.Duty, allEvents map[core.Duty][]event) (bool, c
...
+func analyseParSigs(duty core.Duty, allEvents map[core.Duty][]event) parsigsByMsg {
```
Comment (xenowits) [inline core/tracker/tracker.go:331]: also curious as to why this function needs all the events and not just the events for the duty?
will this work?
```suggestion
func analyseParSigs(duty core.Duty, events []event) parsigsByMsg {
```

**Review by xenowits — APPROVED** [review]

File: core/tracker/tracker.go (hunk 1 above)
Comment (corverroos) [reply core/tracker/tracker.go:765]: no, `SyncCommitteeSelection` should always be consistent, whlie `ContributionAndProof` contains the beaconblockroot that VC pick themselves.

File: core/tracker/tracker.go (hunk 3 above)
Comment (corverroos) [reply core/tracker/tracker.go:331]: indeed, good catch

File: core/tracker/tracker.go (hunk 1 above)
Comment (xenowits) [reply core/tracker/tracker.go:765]: true

**Pull request merged**